tasks/request_files/request_files.py

```diff
@@ -235,88 +259,117 @@ def inner_task(event: Dict, max_retries: int, retry_sleep_secs: float,
-                    'dest_bucket' (str): The bucket the restored file will be moved
-                        to after the restore completes
-                    'success' (bool): Should enter this method set to False. Modified to 'True' by method end.
-                    'err_msg' (str): Will be modified if error occurs.
-
-
-            glacier_bucket: The S3 glacier bucket name.
-            restore_expire_days:
-                The number of days the restored file will be accessible in the S3 bucket before it expires.
-            max_retries: todo
-            retry_sleep_secs: todo
-            retrieval_type: todo
-            db_queue_url: todo
-            job_id: The unique identifier used for tracking requests.
-
-        Raises: RestoreRequestError if any file restore could not be initiated.
+    Args:
+        s3: An instance of boto3 s3 client
+        granule: A dict with the following keys:
+            'granuleId' (str): The id of the granule being restored.
+            'recover_files' (list(dict)): A list of dicts with the following keys:
+                'key' (str): Name of the file within the granule.
+                'dest_bucket' (str): The bucket the restored file will be moved
+                    to after the restore completes
+                'success' (bool): Should enter this method set to False. Modified to 'True' by method end.
+                'err_msg' (str): Will be modified if error occurs.
+
+
+        glacier_bucket: The S3 glacier bucket name.
+        restore_expire_days:
+            The number of days the restored file will be accessible in the S3 bucket before it expires.
+        max_retries: todo
+        retry_sleep_secs: todo
+        retrieval_type: todo
+        db_queue_url: todo
+        job_id: The unique identifier used for tracking requests.
+
+    Raises: RestoreRequestError if any file restore could not be initiated.
     """
-    request_time = datetime.now(timezone.utc).isoformat()
     attempt = 1
     granule_id = granule[GRANULE_GRANULE_ID_KEY]
-
-    post_status_for_job_to_queue(job_id, granule_id, ORCA_STATUS_PENDING, request_time, None, glacier_bucket,
-                                 RequestMethod.POST,
-                                 db_queue_url,
-                                 max_retries, retry_sleep_secs)
+    files = [
+        {
+            "filename": os.path.basename(
+                granule[GRANULE_RECOVER_FILES_KEY][FILE_KEY_KEY]
+            )
+        },
+        {"key_path": "TBD"}, #TBD
+        {"restore_destination": "TBD"}, #TBD
+        {"status_id": shared_recovery.OrcaStatus.PENDING.value},
+        {"error_message": None},
+        {"request_time": datetime.now(timezone.utc).isoformat()},
+        {"last_update": datetime.now(timezone.utc).isoformat()},
+        {"completion_time": datetime.now(timezone.utc).isoformat()},
+    ]
+    shared_recovery.create_status_for_job(
+        job_id, granule_id, glacier_bucket, files , db_queue_url
+    )
```

Comment:
in shared_recovery.create_status_for_job func, we now have "files" which is a list of dict with the following keys,values:

        files: A List of Dicts with the following keys:
            'filename' (str)
            'key_path' (str)
            'restore_destination' (str)
            'status_id' (int)
            'error_message' (str, Optional)
            'request_time' (str)
            'last_update' (str)
            'completion_time' (str, Optional)

I am not sure what should be the "filename", "key_path" and "restore_destination" in the above code. Please review @AndrewDorn @bhazuka

Reply:
From the previous code, looks like os.path.basename(a_file[FILE_KEY_KEY], a_file[FILE_KEY_KEY], and a_file[FILE_DEST_BUCKET_KEY] respectively.

Reply:
changed to the above parameters and it worked